src/agents/factory/agent.factory.ts

```
@@ -0,0 +1,34 @@
+    const agentConfig = { ...config, baseUrl };
+
+    switch (config.provider.toLowerCase()) {
+      case 'hebo':
+        return new HeboAgent(agentConfig);
+      case 'openai':
+        return new OpenAIAgent(agentConfig);
+      default:
+        throw new Error(`Unsupported provider: ${config.provider}`);
+    }
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)